texastribune/ds-toolbox
# Convert to queso-ui npm/asset library

## This can't merge until after https://github.com/texastribune/donations/pull/255

#### What's this PR do?

- Reorganizes this repo as an asset library for npm and an accompanying style documentation utility.
     - `/tasks` => `docs/config/tasks` (to mirror what we do in other repos)
     - npm will ignore the entire `/docs/` folder leaving only `/assets/` available to import
- package.json updates
     - Adds a `yarn release` task for publishing to npm
     - Adds npm info for `@texastribune/queso-ui`
     - Drops most dependencies in favor of just ones needed for sass compiling (the others are purely for building out docs)



#### Why are we doing this? How does it help us?

It's a grand plan to simplify our design system iteration process. Splitting our design system repo into 2 repos helps us directly include our asset library (queso-ui) in /texastribune and include the task runner helpers (queso-tools) in tt-docker-base.

#### How should this be manually tested?

`yarn`
`yarn build` - Everything successfully builds
`yarn dev` - A browserSync server link appears and you can successfully navigate to it.
`yarn lint` - Lints all the things; no errors

#### How should this change be communicated to end users?

End users are us so we should update the group once we set this up in our /donations and /texastribune repos

#### Are there any smells or added technical debt to note?

Not that I can think of

#### What are the relevant tickets?
off sprint, but all part of our CSS refactor


#### queso roll-out plan

* [x] Convert _ds-toolbox_ compiling tasks to queso-tools
* [x] Convert _donations_ compiling tasks to queso-tools
* [x] **Deprecate styles.js and icons.js and related dependencies in _ds-toolbox_** <= we are here
* [x] **Restructure _ds-toolbox_ to form queso-ui package (leaving assets folder intact)** <= and here
* [ ] Convert _donations_ asset grab to queso-ui
* [ ] Convert _texastribune_ asset grab to queso-ui
